src/Converter.cc

```diff
@@ -229,13 +240,304 @@ void Converter::ConvertImpl(tinyxml2::XMLElement *_elem,
+
+    if (elem->Attribute("name"))
+      elemAttrName = elem->Attribute("name");
+
+    if (elemAttrName.empty() ||
+        elemAttrName.compare(0, _newNameIdx - 2, newModelName) != 0 ||
+        (elemName != "frame" && elemName != "joint" &&
+        elemName != "link" && elemName != "model"))
+    {
+      if (elemName != "gripper")
+      {
+        elem = nextElem;
+        continue;
+      }
+    }
+
+    // Child attribute name w/ newModelName prefix stripped except for //gripper
+    std::string childAttrName;
+    if (elemName != "gripper")
+    {
+      childAttrName = elemAttrName.substr(_newNameIdx);
+      elem->SetAttribute("name", childAttrName.c_str());
+    }
+
+    // strip new model prefix from //pose/@relative_to
+    tinyxml2::XMLElement *poseElem = elem->FirstChildElement("pose");
+    if (poseElem != nullptr && poseElem->Attribute("relative_to"))
+    {
+      std::string poseRelTo = poseElem->Attribute("relative_to");
+
+      if (poseRelTo.compare(0, _newNameIdx - 2, newModelName) == 0)
+      {
+        poseRelTo = poseRelTo.substr(_newNameIdx);
+        poseElem->SetAttribute("relative_to", poseRelTo.c_str());
+      }
+    }
+
+    if (elemName == "frame")
+    {
+      std::string attachedTo;
+
+      if (elem->Attribute("attached_to"))
+      {
+        attachedTo = elem->Attribute("attached_to");
+
+        SDF_ASSERT(attachedTo.compare(0, _newNameIdx - 2, newModelName) == 0,
+          "Error: Frame attribute 'attached_to' does not start with " +
+          newModelName);
+
+        // strip new model prefix from attached_to
+        attachedTo = attachedTo.substr(_newNameIdx);
+        elem->SetAttribute("attached_to", attachedTo.c_str());
+      }
+
+      // remove frame if childAttrName == __model__
+      if (childAttrName == "__model__" && elem->Attribute("attached_to"))
+      {
+        _newModel->SetAttribute("canonical_link", attachedTo.c_str());
+        _newModel->InsertFirstChild(poseElem);
+
+        _elem->DeleteChild(elem);
+        elem = poseElem;
+      }
+    }  // frame
+
+    else if (elemName == "link")
+    {
+      // find & strip new model prefix of all //link/<element>/pose/@relative_to
+      for (tinyxml2::XMLElement *e = elem->FirstChildElement();
+          e;
+          e = e->NextSiblingElement())
+      {
+        poseElem = e->FirstChildElement("pose");
+        if (poseElem != nullptr)
+        {
+          std::string poseRelTo = poseElem->Attribute("relative_to");
+
+          SDF_ASSERT(poseRelTo.compare(0, _newNameIdx - 2, newModelName) == 0,
+            "Error: Pose attribute 'relative_to' does not start with " +
+            newModelName);
+
+          poseRelTo = poseRelTo.substr(_newNameIdx);
+          poseElem->SetAttribute("relative_to", poseRelTo.c_str());
+        }
+      }
+    }  // link
+
+    else if (elemName == "joint")
+    {
+      // strip new model prefix from //joint/parent
+      tinyxml2::XMLElement *e = elem->FirstChildElement("parent");
```

Comment:
I think we should check if `e` and  `e->GetText()` are nullptr. Here and for `child`.

Reply:
dce6828593004010ad58c44ab45f8d0b7bf462be